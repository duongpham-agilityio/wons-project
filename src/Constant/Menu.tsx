import dynamic from 'next/dynamic';
import {
  ANALYTICS_PAGE,
  ARCHIVE_PAGE,
  DASHBOARD_PAGE,
  INBOX_PAGE,
  MARKET_PAGE,
  NEWS_FEED_PAGE,
  PROFILE_PAGE,
  SETTINGS_PAGE,
  WALLET_PAGE,
} from './Router';

const Dashboard = dynamic(() => import('@/icons/Dashboard'));
const Inbox = dynamic(() => import('@/icons/Inbox'));
const Wallet = dynamic(() => import('@/icons/Wallet'));
const NewsFeed = dynamic(() => import('@/icons/NewsFeed'));
const Analytics = dynamic(() => import('@/icons/Analytics'));
const Market = dynamic(() => import('@/icons/Market'));
const Archive = dynamic(() => import('@/icons/Archive'));
const Profile = dynamic(() => import('@/icons/Profile'));
const Settings = dynamic(() => import('@/icons/Settings'));

export const MENU = [
  {
    icon: <Dashboard width="30px" height="30px" />,
    label: 'Dashboard',
    className: 'text-blue-400 bg-translucent-400',
    href: DASHBOARD_PAGE,
  },
  {
    icon: <Inbox width="30px" height="30px" />,
    label: 'Inbox',
    href: INBOX_PAGE,
  },
  {
    icon: <Wallet width="30px" height="30px" />,
    label: 'Wallet',
    href: WALLET_PAGE,
  },
  {
    icon: <NewsFeed width="30px" height="30px" />,
    label: 'NewsFeed',
    href: NEWS_FEED_PAGE,
  },
  {
    icon: <Analytics width="30px" height="30px" />,
    label: 'Analytics',
    href: ANALYTICS_PAGE,
  },
  {
    icon: <Market width="30px" height="30px" />,
    label: 'Market',
    href: MARKET_PAGE,
  },
  {
    icon: <Archive width="30px" height="30px" />,
    label: 'Archive',
    href: ARCHIVE_PAGE,
  },
  {
    icon: <Profile width="30px" height="30px" />,
    label: 'Profile',
    className: 'mt-[100px]',
    href: PROFILE_PAGE,
  },
  {
    icon: <Settings width="30px" height="30px" />,
    label: 'Settings',
    href: SETTINGS_PAGE,
  },
];
