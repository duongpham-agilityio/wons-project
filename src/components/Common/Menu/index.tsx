import Link from 'next/link';
import { twMerge } from 'tailwind-merge';
import { MENU } from '@/constant';

const Menu = () => {
  return (
    <nav className="px-6 border-b-2 border-gray-50 dark:border-gray-700">
      <ul>
        {MENU.map(({ icon, label, className, href }) => (
          <Link href={href} key={label}>
            <li
              className={twMerge(
                'flex py-lg pl-2xl rounded-lg items-center cursor-pointer font-semibold text-grayBlue-400',
                className,
              )}
            >
              {icon}
              <span className="ml-5">{label}</span>
            </li>
          </Link>
        ))}
      </ul>
    </nav>
  );
};

export default Menu;
