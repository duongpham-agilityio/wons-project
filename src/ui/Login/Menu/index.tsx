import Link from 'next/link';
import { MENU_NO_AUTH } from '@/constants';

const Menu = () => {
  return (
    <ul className="flex gap-24 items-center">
      {MENU_NO_AUTH.map(({ label, href }) => (
        <li key={label}>
          <Link href={href} className="text-xl font-semibold text-grayBlue-400">
            <span>{label}</span>
          </Link>
        </li>
      ))}
    </ul>
  );
};

export default Menu;
