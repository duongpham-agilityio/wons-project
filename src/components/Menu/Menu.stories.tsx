import { Meta, StoryObj } from '@storybook/react';
import Menu from './index';

export default {
  title: 'Components/Menu',
  component: Menu,
} as Meta;

type Story = StoryObj<typeof Menu>;

// Menu type default
export const MenuDefault: Story = {
  args: {},
};
