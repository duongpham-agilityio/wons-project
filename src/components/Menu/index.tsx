import Link from 'next/link';
import { MENU } from '@/Constant';

const Menu = () => {
  return (
    <nav className="px-6 border-b-2 border-gray-50">
      <ul>
        {MENU.map(({ icon, label, className, href }) => (
          <Link href={href} key={label}>
            <li
              className={`flex py-[18px] pl-[26px] rounded-lg items-center cursor-pointer font-semibold ${className}`}
            >
              {icon}
              <span className="ml-5">{label}</span>
            </li>
          </Link>
        ))}
      </ul>
    </nav>
  );
};

export default Menu;
