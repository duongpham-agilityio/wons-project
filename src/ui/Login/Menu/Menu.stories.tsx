import { Meta, StoryObj } from '@storybook/react';
import Menu from '.';

export default {
  title: 'UI/Login/Menu',
  component: Menu,
} as Meta;

type Story = StoryObj<typeof Menu>;

// Menu type default
export const MenuDefault: Story = {
  args: {},
};
