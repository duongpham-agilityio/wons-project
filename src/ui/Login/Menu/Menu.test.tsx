import { render } from '@testing-library/react';
import Menu from '.';

describe('Menu Component', () => {
  test('render Sidebar', () => {
    const { container } = render(<Menu />);
    expect(container).toMatchSnapshot();
  });
});
