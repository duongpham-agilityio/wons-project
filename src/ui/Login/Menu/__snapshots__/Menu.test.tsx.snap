// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component render Sidebar 1`] = `
<div>
  <ul
    class="flex gap-24 items-center"
  >
    <li>
      <a
        class="text-xl font-semibold text-grayBlue-400"
        href=""
      >
        <span>
          Overview
        </span>
      </a>
    </li>
    <li>
      <a
        class="text-xl font-semibold text-grayBlue-400"
        href=""
      >
        <span>
          Projects
        </span>
      </a>
    </li>
    <li>
      <a
        class="text-xl font-semibold text-grayBlue-400"
        href=""
      >
        <span>
          Tasks
        </span>
      </a>
    </li>
    <li>
      <a
        class="text-xl font-semibold text-grayBlue-400"
        href=""
      >
        <span>
          Reports
        </span>
      </a>
    </li>
    <li>
      <a
        class="text-xl font-semibold text-grayBlue-400"
        href=""
      >
        <span>
          Statements
        </span>
      </a>
    </li>
  </ul>
</div>
`;
